Add intro about MSTest (#40045)
dotnet/docs: docs/core/testing/unit-testing-mstest-intro.md

---
title: MSTest overview
description: Learn about MSTest.
author: Evangelink
ms.author: amauryleve
ms.date: 03/17/2024
---

# MSTest overview

> MSTest, Microsoft Testing Framework, is a test framework for .NET applications. It allows to write and execute tests, tests suites with integration to Visual Studio and Visual Studio Code Test Explorers, .NET CLI, and many CI.

Author: ```suggestion
MSTest, Microsoft Testing Framework, is a test framework for .NET applications. It allows to write and execute tests, tests suites with integration to Visual Studio and Visual Studio Code Test Explorers, .NET CLI, and many CI pipelines.
```

> MSTest, Microsoft Testing Framework, is a test framework for .NET applications. It allows to write and execute tests, tests suites with integration to Visual Studio and Visual Studio Code Test Explorers, .NET CLI, and many CI pipelines.

Reviewer: ```suggestion
MSTest, Microsoft Testing Framework, is a test framework for .NET applications. It allows you to write and execute tests, and provide test suites with integration to Visual Studio and Visual Studio Code Test Explorers, the .NET CLI, and many CI pipelines.
```

> This is a fully supported, open-source and cross-platform test framwork that works with all supported .NET Frameworks (.NET Framwork, .NET Core, .NET, UWP, WinUI...) hosted on [GitHub](https://github.com/microsoft/testfx).

Reviewer: ```suggestion
MSTest is a fully supported, open-source and a cross-platform test framework that works with all supported .NET targets (.NET Framework, .NET Core, .NET, UWP, WinUI, and so on) hosted on [GitHub](https://github.com/microsoft/testfx).
```

## MSTest support policy

Since v3.0.0, MSTest is strictly following [semantic versioning](../../csharp/versioning.md#semantic-versioning).

> MSTest team only supports the latest released version and strongly encourages its users and customers to always update to latest version to benefit from new improvements and security patches. Preview releases aren't supported by Microsoft but they are offered for public testing ahead of the final release.

Reviewer: ```suggestion
The MSTest team only supports the latest released version and strongly encourages its users and customers to always update to latest version to benefit from new improvements and security patches. Preview releases aren't supported by Microsoft, but they are offered for public testing ahead of the final release.
```
